elytron/src/main/java/org/wildfly/extension/elytron/AuthenticationClientDefinitions.java

```diff
@@ -120,12 +120,14 @@
             .build();
 
     static final SimpleAttributeDefinition SECURITY_DOMAIN = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.SECURITY_DOMAIN, ModelType.STRING, true)
+            .setAllowExpression(true)
```

Comment:
This should not be an expression. Model references should not be expressions as they can't be resolved in Stage.MODEL 100% of the time.